Tests/Marketplace/Tests/search_and_install_packs_test.py

```diff
@@ -0,0 +1,113 @@
+import demisto_client
+from Tests.test_content import ParallelPrintsManager
+from Tests.Marketplace.search_and_install_packs import search_and_install_packs_and_their_dependencies
+
+BASE_URL = 'http://123-fake-api.com'
+API_KEY = 'test-api-key'
+
+MOCK_PACKS_SEARCH_RESULTS = """{
+    "packs": [
+        {
+            "id": "HelloWorld",
+            "currentVersion": "2.0.0",
+            "name": "HelloWorld",
+            "dependencies": {"Base": {}, "TestPack": {}}
+        },
+        {
+            "id": "TestPack",
+            "currentVersion": "1.0.0",
+            "name": "TestPack",
+            "dependencies": {"Base": {}}
+        },
+        {
+            "id": "AzureSentinel",
+            "currentVersion": "1.0.0",
+            "name": "AzureSentinel",
+            "dependencies": {"Base": {}}
+        },
+        {
+            "id": "Base",
+            "currentVersion": "1.0.0",
+            "name": "Base",
+            "dependencies": {}
+        }
+    ]
+}"""
+
+MOCK_PACKS_INSTALLATION_RESULT = """[
+    {
+        "id": "HelloWorld",
+        "currentVersion": "2.0.0",
+        "name": "HelloWorldPremium",
+        "installed": "2020-04-06T16:35:10.998538+03:00"
+    },
+    {
+        "id": "TestPack",
+        "currentVersion": "1.0.0",
+        "name": "TestPack",
+        "installed": "2020-04-13T16:43:22.304144+03:00"
+    },
+    {
+        "id": "AzureSentinel",
+        "currentVersion": "1.0.0",
+        "name": "AzureSentinel",
+        "installed": "2020-04-13T16:57:32.655598+03:00"
+    },
+    {
+        "id": "Base",
+        "currentVersion": "1.0.0",
+        "name": "Base",
+        "installed": "2020-04-06T14:54:09.755811+03:00"
+    }
+]"""
+
+
+def mocked_generic_request_func(self, path, method, body, accept):
+    if path == '/contentpacks/marketplace/search':
+        return MOCK_PACKS_SEARCH_RESULTS, 200, None
+    elif path == '/contentpacks/marketplace/install':
+        return MOCK_PACKS_INSTALLATION_RESULT, 200, None
+    return None, None, None
+
+
+class MockConfiguration:
+    def __init__(self):
+        self.host = None
+
+
+class MockApiClient:
+    def __init__(self):
+        self.configuration = MockConfiguration()
+
+
+class MockClient:
+    def __init__(self):
+        self.api_client = MockApiClient()
+
+
+def test_search_and_install_packs_and_their_dependencies(mocker):
```

Comment:
please add documentation to every tests.
you can see the file`demisto_sdk/tests/integration_tests/secrets_integration_test.py` in the pr: https://github.com/demisto/demisto-sdk/pull/360/files